PowerEditor/src/WinControls/DockingWnd/DockingCont.cpp

```diff
@@ -1227,7 +1265,52 @@ intptr_t CALLBACK DockingCont::run_dlgProc(UINT Message, WPARAM wParam, LPARAM l
 			}
 			return FALSE;
 		}
-		case WM_COMMAND : 
+
+		case WM_DPICHANGED:
+		{
+			_dpiManager.setDpiWP(wParam);
+			[[fallthrough]];
+		}
+		case WM_DPICHANGED_AFTERPARENT:
+		{
+			if (Message != WM_DPICHANGED)
+			{
+				_dpiManager.setDpi(_hParent);
+			}
+			_captionHeightDynamic = _dpiManager.scale(HIGH_CAPTION);
+			_captionGapDynamic = _dpiManager.scale(CAPTION_GAP);
+			_closeButtonPosLeftDynamic = _dpiManager.scale(CLOSEBTN_POS_LEFT);
+			_closeButtonPosTopDynamic = _dpiManager.scale(CLOSEBTN_POS_TOP);
+
+			_closeButtonWidth = _dpiManager.scale(g_dockingContCloseBtnSize);
+			_closeButtonHeight = _dpiManager.scale(g_dockingContCloseBtnSize);
+
+			const int tabDpiPadding = _dpiManager.scale(g_dockingContTabIconSize + g_dockingContTabIconPadding * 2);
+			::SendMessage(_hContTab, TCM_SETMINTABWIDTH, 0, tabDpiPadding);
+			TabCtrl_SetPadding(_hContTab, tabDpiPadding / 2, 0);
+			TabCtrl_SetItemSize(_hContTab, 2 * tabDpiPadding, tabDpiPadding);
+
+			destroyFonts();
+
+			LOGFONT lfTab{ _dpiManager.getDefaultGUIFontForDpi() };
+			_hFont = ::CreateFontIndirect(&lfTab);
+
+			LOGFONT lfCaption{ _dpiManager.getDefaultGUIFontForDpi(DPIManagerV2::FontType::smcaption) };
+			_hFontCaption = ::CreateFontIndirect(&lfCaption);
+
+			if (Message == WM_DPICHANGED)
+			{
+				_dpiManager.setPositionDpi(lParam, _hSelf);
+			}
+			else
+			{
+				onSize();
+			}
+
+			return TRUE;

```

Comment:
For [WM_DPICHANGED_AFTERPARENT](https://learn.microsoft.com/en-us/windows/win32/hidpi/wm-dpichanged-afterparent):
```
Return value
This value is unused and ignored by the system.
```
So it doesn't matter the processing this message returns TRUE.
However,  since [WM_DPICHANGED ](https://learn.microsoft.com/en-us/windows/win32/hidpi/wm-dpichanged)use the same return value, there's a problem to me:
```
Return value
If an application processes this message, it should return zero.
```

So I guess it should be `return TRUE;`

Reply:
`DockingCont::run_dlgProc` is dialog procedure. `WM_DPICHANGED` is processed so it should return `TRUE`.